extensions/wikia/WallNotifications/tests/WallNotificationEntityTest.php

```diff
@@ -1,24 +1,29 @@
 <?php
 
-/**
- * Class WallNotificationEntityTest
- */
 class WallNotificationEntityTest extends WikiaBaseTest {
 
+	public function setUp() {
+		$this->setupFile = __DIR__ . '/../WallNotifications.setup.php';
+		parent::setUp();
+	}
+
 	/**
-	 * @param $params
+	 * Given a revision, verify that wall notification data is properly populated
 	 *
+	 * @see WallNotificationEntity::createFromRev()
+	 * @group Slow
+	 * @slowExecutionTime 0.01071 ms
 	 * @dataProvider loadDataFromRevDataProvider
+	 * @param array $params
 	 */
-	public function testLoadDataFromRev( $params ) {
-		F::app()->wg->Sitename = $params['wgSitename'];
-		F::app()->wg->CityId = $params['wgCityId'];
-
-		$rev = $this->getRevisionMock( $params );
-		$entity = $this->getEntityMock( $rev, $params );
+	public function testCreateFromRev( array $params ) {
+		$this->setupEnvironmentMocks( $params );
+		$this->setupWallMessageMocks( $params );
+		$revisionMock = $this->getRevisionMock( $params );
 
-		$entity->loadDataFromRev( $rev );
+		$entity = WallNotificationEntity::createFromRev( $revisionMock );
 
+		$this->assertInstanceOf( WallNotificationEntity::class, $entity, 'WallNotificationEntity::createFromRev must return instance of WallNotificationEntity if given valid revision' );
 		$this->assertEquals( $entity->id, $params['id'], 'Testing "id" matches' );
 		$this->assertEquals( $entity->data, $params['data'], 'Member "data" matches' );
 		$this->assertEquals( $entity->parentTitleDbKey, $params['parentTitleDbKey'], 'Testing "parentTitleDbKey" matches' );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "parentTitleDbKey" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)